pyhap/characteristic.py

```diff
@@ -163,70 +150,45 @@ def override_properties(self, properties=None, valid_values=None):
         if valid_values:
             self.properties['ValidValues'] = valid_values
 
-    def get_hap_value(self):
-        """Get the value of the characteristic, constrained with the HAP properties.
-        """
-        val = self.value
-        if self.properties["Format"] == HAP_FORMAT.STRING:
-            val = val[:256]
-        elif self.properties["Format"] in HAP_FORMAT.NUMERIC:
-            if "maxValue" in self.properties:
-                val = min(self.properties["maxValue"], val)
-            if "minValue" in self.properties:
-                val = max(self.properties["minValue"], val)
-        return val
-
     def notify(self):
-        """Notify clients about a value change.
-
-        .. note:: Non-blocking, i.e. does not wait for the update to be sent.
-        .. note:: Uses the `get_hap_value`, i.e. sends the HAP value.
-        .. seealso:: accessory_driver.publish
-
-        :raise NotConfiguredError: When the broker is not set.
-        """
-        if self.broker is None:
-            raise NotConfiguredError("Attempted to notify when `broker` is None. "
-                                     "Consider adding the characteristic to a "
-                                     "Service and then to an Accessory.")
-
-        data = {
-            "type_id": self.type_id,
-            "value": self.get_hap_value(),
+        """Notify clients about a value change. Sends the value."""
+        self.broker.publish(self.value, self)
+
+    def to_dict(self):
```

Comment:
Not sure about this. See comment in PR.

Reply:
I would prefer `to_HAP` for the same reasons to be honest. If `to_dict` is so needed, it can currently just return the result of `to_HAP`.

Reply:
> to_HAP exists to create a representation of the characteristic that the HAP protocol understands. The fact that it's a dict is just an implementation detail, and it shouldn't be relied on.

I disagree. It might just be an implementation detail, but that exactly makes it even better. `to_HAP` might not be clear for someone without a background in HomeKit who just started looking at `HAP-python`.

> Changing it to to_dict give the impression that you could use that representation for something else

Why not? If someone whats to use that say for debugging, why should we limit that?

> Changing it to to_dict give the impression [...] that it's always going to be a dict type. Neither of those things are true, in the future we may decide to introduce a HAPMessage type or something similar, which would be returned instead of a dict. to_dict ties our hands on that, conflating what the function does with what it returns.

I get that, but what prevents us to changing that back in the future?

---

I believe that `to_dict` would be a better representation of this method since this method name is widely used across many different python projects, including Home Assistant (for `States` and `Events`).
However if you two think otherwise, this shouldn't stand in the way of this PR.

Reply:
You can use `to_HAP` for debugging without it being `to_dict`. How about this:
```python
def to_dict(self):
    return self.to_HAP()
```
In this way, we get best of both worlds. What is more, when/if `to_dict` changes because there are more stuff in the Char we want to dict, we can just add those in `to_dict`

Reply:
Still not convinced by this. `to_HAP` has a very explicit purpose, which is to serialise the object for the HAP proto. 

I agree with @ikalchev, either `to_dict` should return the result of `to_HAP`, or perhaps better, `to_HAP` uses `to_dict` internally to build its serialisation. 

Either way `to_HAP` shouldn't go anywhere as it's an important internal API. But I don't see any issues with added an additional `to_dict` method.

Reply:
I will leave it at `to_HAP`.
```py
def to_dict(self):
    return self.to_HAP()
```
That would add just another method to do the same think and would defeat the propose of this change (to reduce complexity). I didn't thought that would be so controversial :sweat_smile:

Reply:
To nit pick, I think it actually makes more sense to do it the other way around. `to_HAP` calls `to_dict` and then trims the result.

Reply:
What one developer can do in one month, two developers can do in two months.

Reply:
> to_HAP calls to_dict and then trims the result.
Probably better, yes

Reply:
For a representation / debugging: `__repr__` would be the way to go. I would stick with `to_HAP` for now then.